### DISCUSSION TITLE
Derivatives of mj_step with respect to model parameters with MJX

### DISCUSSION THREAD

**Discussion by tombelv** [google-deepmind/mujoco#1618] [Asking for Help]
Hi, 
I'm currently using mujoco for model-based robust control and I'm interested in computing the sensitivities of mj_step with respect to different quantities. In particular, I'm looking for a way to compute derivatives wrt model parameters such as the inertial parameters of the robot links.

Do you have any comment on whether this is even possible given the current implementation, and if yes how could I proceed?

**Comment by antalpeter1** [comment]
Hi,
I would also need this for my project. Did you find out a solution in the meantime?
Peter

**Reply by tombelv** [reply]
Hi Peter, 
unfortunately no, I had to resort to something else.

I wonder however if the new mjSpec API could be exploited to do this, but I haven't had the time to look into it.

**Reply by antalpeter1** [reply]
Actually, I'm using the original C implementation (not MJX), where the gradients of mj_step are computed with finite differences (mjd_transitionFD). Maybe parameter sensitivities could also be computed similarly using FD, by perturbing model parameters through e.g. setting the value of m->body_mass. Do you think this makes sense?

**Reply by yuvaltassa** [reply]
Just finite difference manually, exactly as you propose. It works well.
